src/client/components/mapComponents/Map.jsx

```diff
@@ -0,0 +1,43 @@
+import React from 'react';
+import MapBase from './MapBase';
+import Marker from './Marker';
+import Icons from './Icons';
+import PropTypes from 'prop-types';
+
+const Map = ({ center, markers, currentPosition, style }) => {
+  if (!center) {
+    throw Error('Center position for map is not defined');
+  }
+
+  let markersToRender = markers || [];
+  if (currentPosition) {
+    markersToRender = markersToRender.concat({
```

Comment:
Try to think about your naming here. How can both of these be called markersToRender if they are not these same. The one you assign to must differ in some way

Reply:
Ok. I see your logic now. But it's C programming style. 
I suggest this instead
```
const markersToRender = currentPosition
   ? [...markers, {
      id: null,
      position: 343,
      icon: 234,
    }]
  : markers
```

Reply:
... with the right prop values of course

Reply:
The code you suggested dose not have fall back for markers . so if no fall back are provided the spread would fail.
and about style I can not have any preferences as I have only coded for 3 months or so. It was just spreading it in multi line helps me to understanding my code.
I will updated the code according your suggestion

Reply:
You mean markers could be null or undefined ? In that case markers has to be atleast an empty array when its inserted. This is best practice